scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt

```diff
@@ -1624,25 +1624,48 @@ internal fun getCommonEnvVars(
   return (minimalEnvVars + commonEnvVars).toMutableMap()
 }
 
+private fun getNewDependenciesIfExist(
+    mergedContainerNames: MutableMap<String, String>,
+    container: ScenarioRunContainer
+): ScenarioRunContainer {
+  var tmp: ScenarioRunContainer = container
+  var dependencies: MutableList<String> = mutableListOf()
+  container.dependencies?.forEach {
+    if (mergedContainerNames.containsKey(it)) {
+      dependencies.add(mergedContainerNames.get(it)!!)
+    }
+  }
+  if (dependencies.isNotEmpty()) {
+    tmp = container.copy(dependencies = dependencies)
+  }
+  return tmp
+}
+
 private fun stackSolutionContainers(
     containers: MutableList<ScenarioRunContainer>
 ): MutableList<ScenarioRunContainer> {
   val stackedContainers: MutableList<ScenarioRunContainer> = mutableListOf()
   var stackedContainer: ScenarioRunContainer? = null
   var stackedIndex = 1
+  val mergedContainerNames: MutableMap<String, String> = mutableMapOf()
```

Comment:
Can be rewrite `val mergedContainerNames = mutableMapOf<String, String>()` (same remark for line 1647)